enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -247,7 +250,24 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
     } catch (ComponentLookupException e) {
       throw new EnforcerRuleException("Unable to lookup a component " + e.getMessage(), e);
     } catch (DependencyResolutionException e) {
-
+      Throwable cause = e.getCause();
+      while (cause != null) {
+        if (cause instanceof ArtifactTransferException) {
+          ArtifactTransferException artifactException = (ArtifactTransferException) cause;
+          Artifact artifact = artifactException.getArtifact();
+          DependencyNode dependencyGraph = e.getResult().getDependencyGraph();
+          String pathsToArtifact = findPaths(dependencyGraph, artifact);
```

Comment:
would this work more naturally as an instance method on DependencyGraph?

Reply:
This is not our class. Renamed the variable for clarity.